src/lib/fragments/Accordion.react.js

```diff
@@ -4,42 +4,204 @@ import xtype from 'xtypejs';
+            expandCollapseAllButtonElement.focus();
+            return;
+          }
+        } else if (index > 0) { // focus the content of the previous accordion section if it is open
+          const previousIndex = index - 1;
+          const previousSectionIsOpen = this.state.sectionsOpen[previousIndex];
+          if (previousSectionIsOpen) {
+            const contentAtPreviousIndex =  this.contentRefs[previousIndex].current;
+            if (contentAtPreviousIndex) {
+              contentAtPreviousIndex.focus();
+              return;
+            }
+          } else { // previous section is closed so focus the previous heading
+            newIndex = index - 1;
+          }
+        }
+        break;
+      case EventOrigin.SECTION_CONTENT: // focus the heading at the same level as the content
+        newIndex = index;
+        break;
+      default:
+        break;
+    }
+
+    // focus the heading at level newIndex
+    const headingToFocus =  this.headerRefs[newIndex].current;
+    if (headingToFocus) {
+      headingToFocus.focus();
+    }
+  }
+
+  // This method handles the Down arrow key event for an accordion component
+  // depending on the origin of the event (Show All button, section heading, or section content).
+  handleDownArrow = (index, eventOrigin) => {
+    const numberSections = this.props.accordionHeadings.length;
+    let newIndex = index;
+
+    switch (eventOrigin) {
+      case EventOrigin.SHOW_ALL_BUTTON:
+        newIndex = 0; // go to accordion heading 0
+        break;
+      case EventOrigin.SECTION_HEADING:
+        const sectionIsOpen = this.state.sectionsOpen[index];
+        if (sectionIsOpen) {
+          const content = this.contentRefs[index].current;
+          if (content) {
+            content.focus();
+            return;
+          }
+        } else { // section is closed, go to next heading if there is one
+          newIndex = index + 1;
+        }
+        break;
+      case EventOrigin.SECTION_CONTENT:
+        newIndex = index + 1; // go to next heading if there is one
+        break;
+      default:
+        break;
+    }
+
+    if (newIndex >= 0 && newIndex < numberSections) {
+      const nextHeading = this.headerRefs[newIndex].current;
+      if (nextHeading) {
+        nextHeading.focus();
+        return;
+      }
+    } else if (newIndex >= numberSections) { // focus on the next element on the page
+      const currentHeading = this.headerRefs[index].current;
+
+      const nextElement = this.findFocusableElement(currentHeading, "next");
+      if (nextElement) {
+        nextElement.focus();
+      }
+    }
+  }
+
+  handleKeyEvent = (event, index) => {
+    let eventOrigin;
+
+    if (index === -1) {
+      eventOrigin = EventOrigin.SHOW_ALL_BUTTON;
+    } else if (event.target.className === "govuk-accordion__section-content") {
+      eventOrigin = EventOrigin.SECTION_CONTENT;
+    } else if (event.target.className === "accordion-button govuk-accordion__section-button") {
```

Comment:
These feel a little at risk of the classes on the accordion changing but it makes sense. Might work better if they're defined in a constant/variable further up and used here and in the render? Just an idea though!